godot-macros/src/docs/markdown_converter.rs

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Converts [markdown](https://en.wikipedia.org/wiki/Markdown) to [BBCode](https://en.wikipedia.org/wiki/BBCode).
+use markdown::mdast::Node;
+use markdown::{to_mdast, ParseOptions};
+use std::collections::HashMap;
+
+pub fn to_bbcode(md: &str) -> String {
+    // > to_mdast() never errors with normal markdown so feel free to unwrap
+    let n = to_mdast(md, &ParseOptions::gfm()).unwrap();
+    let definitions = n
+        .children()
+        .unwrap() // root node always has children
+        .iter()
+        .filter_map(|n| match n {
+            Node::Definition(definition) => Some((&*definition.identifier, &*definition.url)),
+            _ => None,
+        })
+        .collect::<HashMap<_, _>>();
+
+    walk_node(&n, &definitions).unwrap_or_default()
+}
+
+// will change to &mut String if necessary
```

Comment:
I think this comment isn't needed anymore -- and if it is, it should be more precise.

Please also use full sentences in comments, i.e. start with uppercase + end with period.